iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java

```diff
@@ -96,17 +101,19 @@ private void checkEnvironmentVariableFrom(BlockObject root, BlockObject envFrom,
    * @param creator A function used to wrap the resulting {@link YamlTree} into the corresponding {@link AbstractVariable} class instance.
    */
   private void checkMapResource(BlockObject root, BlockObject mapRef, Container container, Class<? extends MapResource> clazz,
-    Function<YamlTree, ? extends AbstractVariable> creator) {
+    TriFunction<YamlTree, String, YamlTree, ? extends AbstractVariable> creator) {
     retrieveMapRefTree(mapRef).ifPresent((ScalarTree mapRefNameValueTree) -> {
       var mapName = mapRefNameValueTree.value();
       getGlobalResources(root).stream()
         .filter(clazz::isInstance)
         .map(clazz::cast)
```

Comment:
E.g. this looks like a candidate for extraction, those 3 lines use `root` & `clazz` - you can replace 2 parameters with one - it can be some `Supplier` to calculate the stream when needed.